test/LinearCurve.test.js

```diff
@@ -0,0 +1,38 @@
+/* global require beforeEach describe it ethers */
+
+const { assert } = require('chai')
+
+describe('LinearCurve', () => {
+  let linearCurve
+
+  beforeEach(async () => {
+    const LinearCurve = await ethers.getContractFactory('LinearCurve')
+    linearCurve = await LinearCurve.deploy()
+  })
+
+  it('returns the correct price', async () => {
+    async function price (supply, amount) {
+      const bigSupply = ethers.constants.WeiPerEther.mul(supply)
+      const bigAmount = ethers.constants.WeiPerEther.mul(amount)
+      return Number(await linearCurve.price(bigSupply, bigAmount))/10**18
```

Comment:
Why do we divide by 10^18 here? Is it so our assertions are in regular old javascript numbers?

Reply:
I think I'd rather in general say in the big number world for all our math and checking. There are fewer places for problems to crop up in the hand off, and I remember reading somewhere that it's best to just avoid javascript numbers completely for this stuff and use string representations to go into and out of the bigNumber math world.